native/libc.c

```diff
@@ -224,48 +224,51 @@ func_regex_match(PyObject *self, PyObject *args) {
   }
 
   regex_t pat;
-  regmatch_t m[2];
 
   // Should have been checked by regex_parse
   if (regcomp(&pat, pattern, REG_EXTENDED) != 0) {
     fprintf(stderr, "Invalid regex at runtime\n");
     return PyLong_FromLong(-1);
   }
 
-  int ret;
-  // must match at pos 0
-  if (regexec(&pat, str, 2, m, 0) == 0) {
-    debug("MATCH\n");
-  //if (regexec(&pat, str, 2, m, 0) == 0 && !m[0].rm_so) {
-    // Return first parenthesized subexpression as string, or length of match
-
-    /*
-    if (pat.re_nsub>0) {
-      ret->s = xmprintf("%.*s", m[1].rm_eo-m[1].rm_so, target+m[1].rm_so);
-      if (TT.refree) free(TT.refree);
-      TT.refree = ret->s;
-    } else assign_int(ret, m[0].rm_eo);
-    */
-    ret = 1;
 
+
+  PyObject *ret = PyList_New(pat.re_nsub+1);
+
+  if (ret == NULL) {
+    regfree(&pat);
+    return NULL;
+  }
+
+  int outlen = pat.re_nsub + 1;
+  int nomatch = 0;
```

Comment:
nitpick: can we make it `match = 0` and then set it inside the if, and reverse the last test?    I don't like bools with negatives in their names; it leads to double negatives like `!nofoo`.